### DISCUSSION TITLE
Optitrack to ROS(Natnet)

### DISCUSSION THREAD

**Discussion by WoodyloveBo** [bitcraze/discussions#1068] [Q&A]
I am trying to control a drone by sending data to ROS using a motion capture camera Optitrack. I looked it up and found that I can send data to ROS using a ROS driver called Natnet. But I'm not sure how to build this in a WSL environment. I would appreciate it if you could tell me the specific setting method.

**Comment by knmcguire** [comment]
That is outside of our scope. We don't have that much experience with natnet itself. You could try this example of our crazyflie python library but you need to then write the ROS2 wrapper around it yourself: https://github.com/bitcraze/crazyflie-lib-python/blob/master/examples/mocap/mocap_hl_commander.py

Have tried the [Crazyswarm2 ROS2 package](https://imrclab.github.io/crazyswarm2/)? They have optitrack support too.

**Comment by sesem738** [comment]
Hi,

I have successfully made Optitrack data available on ROS using VRPN. I followed this tutorial, which is a bit outdated since it was designed for Motive 2 and ROS Kinetic, but it is not so different that you cannot adapt it: https://tuw-cpsg.github.io/tutorials/optitrack-and-ros/

**Reply by WoodyloveBo** [reply]
I followed that tutorial, but got the following error: I don't know what the problem is.
![캡처](https://github.com/bitcraze/discussions/assets/143582301/82519d00-b848-441f-85b6-09b92a380467)

**Comment by knmcguire** [comment]
I would also advise you to take a look at https://github.com/IMRCLab/motion_capture_tracking/tree/ros2. That is updated for ROS2 now
